pkg/storage/formattedstore/formattedstore.go

```diff
@@ -108,10 +138,26 @@ func (s *formatStore) Put(k string, v []byte) error {
 		return fmt.Errorf(failPutInUnderlyingStore, err)
 	}
 
+	// put data in cache to retrieve it from Get
+	if s.cacheStore != nil {
+		if err := s.cacheStore.Put(k, v); err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 func (s *formatStore) Get(k string) ([]byte, error) {
+	if s.cacheStore != nil {
+		v, err := s.cacheStore.Get(k)
```

Comment:
If the data isn't here in the cache, you'll need to check the non-cache provider

Reply:
(if you're using a mem provider, this will fail if/when the aries agent gets instantiated again)

Reply:
(discussed this offline - this choice is left to the user)